stubs/psycopg2/psycopg2/extras.pyi

```diff
@@ -28,22 +29,42 @@ from psycopg2._range import (
 
 from .extensions import connection as _connection, cursor as _cursor, quote_ident as quote_ident
 
+_T_cur = TypeVar("_T_cur", bound=_cursor)
+
 class DictCursorBase(_cursor):
     row_factory: Any
     def __init__(self, *args, **kwargs) -> None: ...
     def fetchone(self) -> tuple[Any, ...] | None: ...
     def fetchmany(self, size: int | None = ...) -> list[tuple[Any, ...]]: ...
     def fetchall(self) -> list[tuple[Any, ...]]: ...
-    def __iter__(self): ...
+    def __next__(self) -> tuple[Any, ...]: ...
```

Comment:
What is the reason for this change? Both `__iter__` and `__next__` are inherited from `cursor`, so we don't need them here.

Reply:
Indeed, you're right. in fact `fetchone`, `fetchmany`, `fetchall` and `row_factory` are all superfluous too. I'll clean them up.